### PR TITLE
Swap to Duration::round from temporal_rs

### PR DESCRIPTION
**Pull request by robot-head**
This Pull Request fixes part of #{11}.

It changes the following:

- Swaps Duration::round implementation to `temporal_rs`'s

### CODE REVIEW

**Comment by robot-head** [comment]
Paired with https://github.com/boa-dev/temporal/pull/34 gets `Duration.round` conformance up to 27.93%
```
Suite test262\test/built-ins/Temporal/Duration/prototype/round results: total: 111, passed: 31, ignored: 0, failed: 80 , conformance: 27.93%


Results (ECMAScript Next):
Total tests: 111
Passed tests: 31
Ignored tests: 0
Failed tests: 80 (0 panics)
Conformance: 27.93%
```

**Review by nekevss — CHANGES_REQUESTED** [review]
Nice work on this overall!

Had a couple suggestions and a nitpick 😄

File: core/engine/src/builtins/temporal/mod.rs [hunk 1]
```diff
@@ -238,14 +242,27 @@ pub(crate) fn _iterator_to_list_of_types(
...
-    _context: &mut Context,
-) -> JsResult<(Option<PlainDate>, Option<ZonedDateTime>)> {
-    Err(JsNativeError::range()
-        .with_message("not yet implemented.")
-        .into())
+    options: &JsObject,
+    context: &mut Context,
+) -> RelativeTemporalObjectResult {
+    let relative_to = options.get(PropertyKey::from(utf16!("relativeTo")), context)?;
+    let plane_date = match relative_to {
```
[the remark this answers is not in the document]
Comment (robot-head) [reply core/engine/src/builtins/temporal/mod.rs:256]: oops! lol

File: core/engine/src/builtins/temporal/duration/mod.rs [hunk 2]
```diff
@@ -652,15 +654,15 @@ impl Duration {
...
 
         // 13. Let roundingIncrement be ? ToTemporalRoundingIncrement(roundTo).
-        let _rounding_increment = get_temporal_rounding_increment(&round_to, context)?;
+        let rounding_increment = get_temporal_rounding_increment(&round_to, context)?;
 
         // 14. Let roundingMode be ? ToTemporalRoundingMode(roundTo, "halfExpand").
-        let _rounding_mode = get_option(&round_to, utf16!("roundingMode"), context)?
-            .unwrap_or(RoundingMode::HalfExpand);
+        let rounding_mode: Option<TemporalRoundingMode> =
+            get_option::<RoundingMode>(&round_to, utf16!("roundingMode"), context)?.map(Into::into);
```
[the remark this answers is not in the document]
Comment (robot-head) [reply core/engine/src/builtins/temporal/duration/mod.rs:665]: Done

File: core/engine/src/builtins/temporal/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (robot-head) [reply core/engine/src/builtins/temporal/mod.rs]: Done

**Review by nekevss — APPROVED** [review]

**Review by jedel1043 — APPROVED** [review]

**Pull request merged**